enterprise/server/saml/saml.go

```diff
@@ -55,56 +54,37 @@ var (
 type SAMLAuthenticator struct {
 	env           environment.Env
 	samlProviders map[string]*samlsp.Middleware
-	fallback      interfaces.Authenticator
 }
 
-func Register(env environment.Env) error {
-	if *certFile == "" || env.GetAuthenticator() == nil {
-		return nil
+func SamlEnabled(env environment.Env) bool {
+	if (*certFile == "" && *cert == "") || env.GetAuthenticator() == nil {
+		return false
 	}
 	if _, ok := env.GetAuthenticator().(*nullauth.NullAuthenticator); ok {
-		return nil
+		return false
 	}
-	log.Info("SAML auth configured.")
-	env.SetAuthenticator(NewSAMLAuthenticator(env, env.GetAuthenticator()))
-	return nil
+	return true
 }
 
-func NewSAMLAuthenticator(env environment.Env, fallback interfaces.Authenticator) *SAMLAuthenticator {
+func NewSAMLAuthenticator(env environment.Env) *SAMLAuthenticator {
 	return &SAMLAuthenticator{
 		env:           env,
 		samlProviders: make(map[string]*samlsp.Middleware),
-		fallback:      fallback,
 	}
 }
 
-func (a *SAMLAuthenticator) AdminGroupID() string {
-	return a.fallback.AdminGroupID()
-}
-
-func (a *SAMLAuthenticator) AnonymousUsageEnabled() bool {
-	return a.fallback.AnonymousUsageEnabled()
-}
-
-func (a *SAMLAuthenticator) PublicIssuers() []string {
-	return a.fallback.PublicIssuers()
-}
-
 func (a *SAMLAuthenticator) SSOEnabled() bool {
-	if *certFile != "" {
-		return true
-	}
-	return a.fallback.SSOEnabled()
+	return (*certFile == "" && *cert == "")
 }
```

Comment:
haven't fully thought it through but will this method always return `true` ? since `Enabled()` returns false if these are both empty and we don't create a SAMLAuthenticator if Enabled() returns false.

Reply:
Yeah, done